sdk/src/ingredient.rs

```diff
@@ -1459,6 +1459,38 @@ impl std::fmt::Display for Ingredient {
     }
 }
 
+impl TryFrom<serde_json::Value> for Ingredient {
+    type Error = crate::Error;
+
+    fn try_from(value: serde_json::Value) -> std::result::Result<Self, Self::Error> {
+        serde_json::from_value(value).map_err(crate::Error::JsonError)
+    }
+}
+
+impl TryFrom<&str> for Ingredient {
+    type Error = crate::Error;
+
+    fn try_from(json: &str) -> std::result::Result<Self, Self::Error> {
+        Self::from_json(json)
+    }
+}
+
+impl TryFrom<String> for Ingredient {
+    type Error = crate::Error;
+
+    fn try_from(json: String) -> std::result::Result<Self, Self::Error> {
+        Self::from_json(&json)
+    }
+}
+
+impl TryFrom<&String> for Ingredient {
+    type Error = crate::Error;
+
+    fn try_from(json: &String) -> std::result::Result<Self, Self::Error> {
+        Self::from_json(json)
+    }
+}
+
```

Comment:
You may be able to do `impl TryFrom<AsRef<str>>` rather than 3 implementations. I'm not too sure how I feel about this either, I think it makes more sense to have an explicit function to convert from json to an `Ingredient`.